src/utils/viewer/ObjectPickingHelper.cpp

```diff
@@ -0,0 +1,132 @@
+#include "ObjectPickingHelper.h"
+#include <Corrade/Containers/StridedArrayView.h>
+#include <Corrade/Utility/Assert.h>
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/Renderbuffer.h>
+#include <Magnum/GL/RenderbufferFormat.h>
+#include <Magnum/Image.h>
+#include <Magnum/Magnum.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Shaders/Generic.h>
+#include <Magnum/Shaders/MeshVisualizer.h>
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+using Mn::Math::Literals::operator""_rgbf;
+
+ObjectPickingHelper::ObjectPickingHelper(Mn::Vector2i viewportSize) {
+  // create the framebuffer and set the color attachment
+  recreateFramebuffer(viewportSize).mapForDraw();
+  CORRADE_INTERNAL_ASSERT(
+      selectionFramebuffer_.checkStatus(Mn::GL::FramebufferTarget::Draw) ==
+      Mn::GL::Framebuffer::Status::Complete);
+
+  // create a shader
+  Mn::Shaders::MeshVisualizer3D::Flags flags =
+      Mn::Shaders::MeshVisualizer3D::Flag::Wireframe;
+  shader_ = std::make_unique<Mn::Shaders::MeshVisualizer3D>(flags);
```

Comment:
The `shader_` wouldn't need to be a pointer at all, and you could initialize it directly in the class definition in the header:

```cpp
Mn::Shaders::MeshVisualizer3D shader_{Mn::Shaders::MeshVisualizer3D::Flag::Wireframe};
```

This is only a lot of unnecessarily verbose code.

Reply:
I did not realize the flag of the shader is actually fixed before coding up everything. Will change.